src/rust/ide/lib/view/src/graph_editor/builtin/visualization/native/documentation.rs

```diff
@@ -0,0 +1,184 @@
+use crate::graph_editor::component::visualization::DataError;
+use crate::graph_editor::component::visualization::Definition;
+
+use enso_frp as frp;
+use ensogl::display::DomSymbol;
+use ensogl::display::scene::Scene;
+use ensogl::display;
+use ensogl::system::web;
+use ensogl::system::web::StyleSetter;
+use ast::prelude::FallibleResult;
+
+
+// =================
+// === Constants ===
+// =================
+
+pub const DOC_VIEW_WIDTH  : f32 = 300.0;
+pub const DOC_VIEW_MARGIN : f32 = 15.0;
+
+const CORNER_RADIUS   : f32  = crate::graph_editor::component::node::CORNER_RADIUS;
+/// Content in the documentation view when the data is yet to be received.
+const PLACEHOLDER_STR : &str = "<h3>Enso Documentation Viewer</h3><p>No documentation available</p>";
+
+
+/// Generates documentation view stylesheet.
+pub fn get_doc_style() -> String {
+    format!("<style>{}</style>", include_str!("documentation/style.css"))
+}
+
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct ViewModel {
+    logger : Logger,
+    dom    : DomSymbol,
+    size   : Rc<Cell<Vector2>>,
+}
+
+impl ViewModel {
+    /// Constructor.
+    fn new(scene:&Scene) -> Self {
+        let logger          = Logger::new("DocumentationView");
+        let div             = web::create_div();
+        let dom             = DomSymbol::new(&div);
+        let screen          = scene.camera().screen();
+        let doc_view_height = screen.height - (DOC_VIEW_MARGIN * 2.0);
+        let size_vec        = Vector2(DOC_VIEW_WIDTH,doc_view_height);
+        let size            = Rc::new(Cell::new(size_vec));
+
+        dom.dom().set_style_or_warn("white-space"     ,"normal"                        ,&logger);
+        dom.dom().set_style_or_warn("overflow-y"      ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("overflow-x"      ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("background-color","rgba(255, 255, 255, 0.85)"     ,&logger);
+        dom.dom().set_style_or_warn("padding"         ,"5px"                           ,&logger);
+        dom.dom().set_style_or_warn("pointer-events"  ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("border-radius"   ,format!("{}px", CORNER_RADIUS)  ,&logger);
+        dom.dom().set_style_or_warn("width"           ,format!("{}px", DOC_VIEW_WIDTH) ,&logger);
+        dom.dom().set_style_or_warn("height"          ,format!("{}px", doc_view_height),&logger);
+
+        scene.dom.layers.main.manage(&dom);
+        ViewModel {dom,logger,size}.init()
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    /// Sets size of the documentation view.
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    /// Generates HTML documentation for documented suggestion.
+    fn gen_doc(doc: String) -> FallibleResult<String> {
```

Comment:
naming doesnt make sense here. Plus there should not be space after `:`. Please browse all files and all new functions and check all names very carefully. Im not adding new comments about it anymore. I'd love to review it once again after the names will be fixed. Here, for example, the function is named `gen_doc`. Which means it generates a doc, but it consumes `doc` as first argument. So something is bad. Moreover, the documentation tells something about "documented suggestion" which is not explained anywhere in the code and I don't see any connection to the signature / code in the function. All docs should be very clear and connected to the code.